tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java

```diff
@@ -47,153 +46,145 @@
+        resultOne = BCompileUtil.compile("test-src/annotations/annot_access.bal");
+        Assert.assertEquals(resultOne.getErrorCount(), 0);
+
+        resultTwo = BCompileUtil.compile("test-src/annotations/annot_access_with_source_only_points.bal");
+        Assert.assertEquals(resultTwo.getErrorCount(), 0);
+
+        resultThree = BCompileUtil.compile("test-src/annotations/annotations_constant_propagation.bal");
+        Assert.assertEquals(resultThree.getErrorCount(), 0);
+
+        resultFour = BCompileUtil.compile("test-src/annotations/annot_availability.bal");
+        Assert.assertEquals(resultFour.getErrorCount(), 0);
+
+        resultAccessNegative = BCompileUtil.compile("test-src/annotations/annotation_access_negative.bal");
+
+        readOnlyValues = BCompileUtil.compile("test-src/annotations/annotation_readonly_types.bal");
+    }
+
+    @Test(description = "test accessing source only annots at runtime, the annots should not be available",
+            dataProvider = "annotAccessWithSourceOnlyPointsTests")
+    public void testSourceOnlyAnnotAccess(String testFunction) {
+        BValue[] returns = BRunUtil.invoke(resultTwo, testFunction);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertSame(returns[0].getClass(), BBoolean.class);
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+    }
+
+    @Test(description = "Test if constants used in annotation are replaced on constant propagation phase",
+          enabled = false)
+    public void testConstantPropagationOnAnnotation() {
+        BLangPackage pkg = (BLangPackage) resultThree.getAST();
+
+        BLangService service = pkg.services.get(0);
+        BLangRecordLiteral serviceAnnotation = (BLangRecordLiteral) service.annAttachments.get(0).expr;
+        Assert.assertTrue(((BLangRecordLiteral.BLangRecordKeyValueField) serviceAnnotation.fields.get(0)).valueExpr
+                instanceof BLangConstRef);
+
+        BLangRecordLiteral helloFunctionAnnotation = (BLangRecordLiteral) service.resourceFunctions.get(0)
+                .annAttachments.get(0).expr;
+        Assert.assertTrue(((BLangRecordLiteral.BLangRecordKeyValueField) helloFunctionAnnotation.fields.get(0))
+                .valueExpr instanceof BLangConstRef);
+
+    }
+
+    @DataProvider(name = "annotAccessTests")
+    public Object[][] annotAccessTests() {
+        return new Object[][]{
+                { "testTypeAnnotAccess1" },
+                { "testTypeAnnotAccess2" },
+                { "testObjectTypeAnnotAccess1" },
+                { "testObjectTypeAnnotAccess2" },
+                { "testObjectTypeAnnotAccess3" },
+                { "testServiceAnnotAccess1" },
+                { "testServiceAnnotAccess2" },
+                { "testServiceAnnotAccess3" },
+                { "testServiceAnnotAccess4" },
+                { "testFunctionAnnotAccess1" },
+                { "testFunctionAnnotAccess2" }
+                // { "testInlineAnnotAccess" } // TODO: #17936
```

Comment:
https://github.com/ballerina-platform/ballerina-lang/issues/17936 is now fixed, we should be able to fix and enable this.

Reply:
Fixed.